src/build/_builder.py

```diff
@@ -225,6 +227,25 @@ def get_requires_for_build(
         with self._handle_backend(hook_name):
             return set(get_requires(config_settings))
 
+    def filter_variants(
+        self,
+        dependencies: set[str],
+        config_settings: ConfigSettings | None = None,
+    ) -> set[str]:
```

Comment:
This logic is probably right; a docstring or some code comments will be helpful before merging though. Just from reading the code it's hard to tell, and navigating to `evaluate_variant_requirements` loses context.

Reply:
Capturing in `filter_variants` that this is demo-only and specific to `pypa/build` and probably not quite suitable for a final design will be useful here - at least for me, it's the kind of thing I'd forget over time.